### ISSUE TITLE
A bug?? -> .NET 8 - the page 'error' in the new Blazor Web App (WebAssembly) redirects to 404

### ISSUE THREAD

**Issue by DillionVVV** [dotnet/aspnetcore#54238]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

I create a .NET 8 Blazor Web App (WebAssembly) application, and directly open the “/Error” page, first it will show the error page, and then it will redirect to the 404 error and show the Not found.

This issue happens on Microsoft Visual Studio Community 2022 (64-bit) Version 17.9.0 + version and Microsoft Visual Studio Community 2022 (64-bit) - Preview Version 17.10.0 Preview 1.0.

I also test it on Visual Studio Community 2022 Preview version 17.8.0 Preview 7.0 Version, in this version, the Error page located in the .client project, so it will show the error page content if open "/Error" page, without redirect.

### Expected Behavior

If I open the "/Error" page, it should display the error page, not the "Not found".

### Steps To Reproduce

1. Create a new project and select the "**Blazor Web App**" Template.
2. Select the "**WebAssembly**" interactive render mode and select the "**Global**" interactivity location.
![image](https://github.com/dotnet/aspnetcore/assets/159244810/483897f1-3466-42e8-adc1-e9cf70ef70b4)
3. Then, you can see the Error page in the server project, instead of the .client project.
![image](https://github.com/dotnet/aspnetcore/assets/159244810/301e2ce3-5cff-4e08-9a3f-59e8ea29b677)
4.Running the application, and open the "/Error" page, it will show the Error page first, and then display the not found. like this: 
![1](https://github.com/dotnet/aspnetcore/assets/159244810/bb2f6b98-ef1b-4025-988b-67a0e36136b3)

### .NET Version

.NET 8.0.200

### Anything else?

Microsoft Visual Studio Community 2022 (64-bit) - Current Version 17.9.1.

**Comment by DillionVVV** [comment]
Yes, I know that. But since I've chosen to create my project using the "WebAssembly" interactive rendering model and the “Global" Interactivity location, why not it directly puts the error page in the client project?

**Comment by paleocomburo** [comment]
I have a similar issue. I create a WebAssembly (global) web app, and want a custom 404 page. The NotFound tag no longer works for some reason, so I followed the guidance to use `app.UseStatusCodePagesWithReExecute("/StatusCode/{0}");` on the hosting server. Then when you go to a page that doesn't exist I see a flash of the customer error page, and then it is replaced by just a plain text "Not Found" message. I have no idea where this comes from, and how to stop it from happening.
I even added the custom 404 page to both the server and client projects, but then the application gives me an error that I have duplicate routes configured. And I don't understand where that comes from. Do the client and server apps share routes?
I then removed the custom page from the server app (it didn't work there anyway), and added it to the client app. But I still get the same behaviour.
